go-controller/pkg/config/config.go

```diff
@@ -547,6 +553,37 @@ func buildKubernetesConfig(exec kexec.Interface, cli, file *config, defaults *De
 		Kubernetes.CACert = getOVSExternalID(exec, "k8s-ca-certificate")
 	}
 
+	// values for token, cacert, kubeconfig, api-server may be found in several places.
+	// Take the first found when looking in this order: command line options, environment variables, files
```

Comment:
Looks like this should really be:

// Precedence is: command line options, config file, /var/run/secrets/kubernetes.io/serviceaccount/ files, environment variables, defaults

Reply:
@dcbw env comes ahead of service account files. The tests I did show that. What would you like changed?

Reply:
Misread, you are correct.

Reply:
But also, per the code, the precedence is this now: CLI > file > env > SA.

eg the comments are still incorrect because config file overrides env & SA currently in the code.